doc/api/worker.md

```diff
@@ -0,0 +1,357 @@
+
+Workers can also, unlike child processes, share memory efficiently by
+transferring `ArrayBuffer` instances or sharing `SharedArrayBuffer` instances
+between them.
+
+## Example
+
+```js
+const { Worker, isMainThread, postMessage, workerData } = require('worker');
+
+if (isMainThread) {
+  module.exports = async function parseJSAsync(script) {
+    return new Promise((resolve, reject) => {
+      const worker = new Worker(__filename, {
+        workerData: script
+      });
+      worker.on('message', resolve);
+      worker.on('error', reject);
+      worker.on('exit', (code) => {
+        if (code !== 0)
+          reject(new Error(`Worker stopped with exit code ${code}`));
+      });
+    });
+  };
+} else {
+  const { parse } = require('some-js-parsing-library');
+  const script = workerData;
+  postMessage(parse(script));
+}
+```
+
+## Event: 'workerMessage'
+<!-- YAML
+added: REPLACEME
+-->
+
+If this thread was spawned by a `Worker`, this event emits messages sent using
+[`worker.postMessage()`][]. See the [`port.on('message')`][] event for
+more details.
+
+Listeners on this event will receive a clone of the `value` parameter as passed
+to `postMessage()` and no further arguments.
+
+*Note*: This event is emitted on the `worker` module as returned by
+`require('worker')` itself.
+
+## worker.isMainThread
+<!-- YAML
+added: REPLACEME
+-->
+
+* {boolean}
+
+Is `true` if this code is not running inside of a [`Worker`][] thread.
+
+## worker.postMessage(value[, transferList])
+<!-- YAML
+added: REPLACEME
+-->
+
+This method is available if this is a `Worker` thread, which can be tested
+using [`require('worker').isMainThread`][].
+
+Send a message to the parent thread’s `Worker` instance that will be received
+via [`worker.on('message')`][]. See [`port.postMessage()`][] for
+more details.
+
+## worker.threadId
+<!-- YAML
+added: REPLACEME
+-->
+
+* {integer}
+
+An integer identifier for the current thread. On the corresponding worker object
+(if there is any), it is available as [`worker.threadId`][].
+
+## worker.workerData
+<!-- YAML
+added: REPLACEME
+-->
+
+An arbitrary JavaScript value that contains a clone of the data passed
+to this thread’s `Worker` constructor.
+
+
+## Class: MessageChannel
+<!-- YAML
+added: REPLACEME
+-->
+
+The `MessageChannel` has no methods of its own. `new MessageChannel()`
+yields an object with `port1` and `port2` properties, which refer to linked
+[`MessagePort`][] instances.
+
+```js
+const { MessageChannel } = require('worker');
+
+const { port1, port2 } = new MessageChannel();
+port1.on('message', (message) => console.log('received', message));
+port2.postMessage({ foo: 'bar' });
+// prints: received { foo: 'bar' }
+```
+
+## Class: MessagePort
+<!-- YAML
+added: REPLACEME
+-->
+
+* Extends: {EventEmitter}
+
+Instances of the `worker.MessagePort` class represent an asynchronous
+communications channel. It can be used to transfer structured data,
```

Comment:
s/an asynchronous communications channel/an end(or port?) of a two-way asynchronous communications channel/